File: main/src/main/java/com/sedmelluq/discord/lavaplayer/container/playlists/HlsStreamSegment.java
package com.sedmelluq.discord.lavaplayer.container.playlists;

public class HlsStreamSegment {
    /**
     * URL of the segment.
     */
    public final String url;
    /**
     * Duration of the segment in milliseconds. <code>null</code> if unknown.
     */
    public final Long duration;
    /**
     * Name of the segment. <code>null</code> if unknown.
     */
    public final String name;

    public HlsStreamSegment(String url, Long duration, String name) {
        this.url = url;
        this.duration = duration;
        this.name = name;
    }
}

File: main/src/main/java/lavaplayer/container/playlists/HlsStreamSegment.kt
package lavaplayer.container.playlists

data class HlsStreamSegment(
    /**
     * URL of the segment.
     */
    @JvmField val url: String,
    /**
     * Duration of the segment in milliseconds. `null` if unknown.
     */
    @JvmField val duration: Long?,
    /**
     * Name of the segment. `null` if unknown.
     */
    @JvmField val name: String?
)
